### PR TITLE
Add vertical lines to files/folders in tree

### PR DESCRIPTION
**Pull request by kaiyue0329**
Solves first half of issue: #6586 

#### What it does
- When a file is highlighted, add vertical lines for all its sibling files/folders in the tree
- When a folder is highlighted/expanded, add vertical lines for all its child files/folders in the tree

#### How to test
Please refer to the first screenshot in issue: #6586 for steps to test

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/core/src/browser/style/tree.css [hunk 1]
```diff
@@ -79,7 +79,7 @@
     font-family: FontAwesome;
     font-size: calc(var(--theia-content-font-size) * 0.8);
     content: "\f0da";
-    transform: rotate(45deg);
+    transform: rotate(90deg);
```
Comment (vince-fugnitto) [inline packages/core/src/browser/style/tree.css]: @kaiyue0329 do you mind explaining why this change is necessary?

File: packages/core/src/browser/style/tree.css [hunk 2]
```diff
@@ -139,3 +139,11 @@
 .theia-tree-element-node {
     width: 100%
 }
+
+.theia-tree-folder-vertical-line .theia-TreeNodeContent {
+    border-left: 1px solid #f0f0f0;
```
Comment (vince-fugnitto) [inline packages/core/src/browser/style/tree.css]: We should not use hardcoded color values, the following fails to display on light-themes:

Screen Shot 2020-01-17 at 4 39 18 PM

Instead, we should use a builtin color which is updated per theme.
You can also take a look at #6475  for reference.

File: packages/core/src/browser/tree/tree-widget.tsx [hunk 3]
```diff
@@ -844,7 +844,7 @@ export class TreeWidget extends ReactWidget implements StatefulWidget {
     }
 
     protected getPaddingLeft(node: TreeNode, props: NodeProps): number {
-        return props.depth * this.props.leftPadding + (this.needsExpansionTogglePadding(node) ? this.props.expansionTogglePadding : 0);
```
Comment (vince-fugnitto) [inline packages/core/src/browser/tree/tree-widget.tsx]: I believe this change will not work once we have the preference to control the behavior of the indent guide.

Comment (akosyakov) [reply packages/core/src/browser/tree/tree-widget.tsx]: I’m pretty sure it breaks some trees or themes. If we change something in general tree, we have to retest all possible combinations of trees and file icon themes, i.e. default, none, seti (font based), minimal (image based(.

File: packages/navigator/src/browser/navigator-widget.tsx [hunk 4]
```diff
@@ -67,22 +67,23 @@ export class FileNavigatorWidget extends FileTreeWidget {
         this.addClass(CLASS);
     }
 
+    protected nodesToAddVerticalLine = new Set<string>();
```
Comment (akosyakov) [inline packages/navigator/src/browser/navigator-widget.tsx]: Why do we need it? Keeping a separate state from main state is common cause of race conditions. Can we decide which node should have vertical line in render function?

**Comment by akosyakov** [comment]
Should not we rather implement indent div blocks for each level? And then be able to style them based on preferences? I don’t understand why do we need any additional state for it?

File: packages/core/src/browser/style/tree.css (hunk 1 above)
Comment (kaiyue0329) [reply packages/core/src/browser/style/tree.css]: I changed it so that the arrow can point to the vertical line when a folder is expanded in the tree.

File: packages/core/src/browser/style/tree.css (hunk 2 above)
Comment (vince-fugnitto) [reply packages/core/src/browser/style/tree.css]: You can take a look at the new coding guidelines regarding theming + colors: https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#theming

Comment (vince-fugnitto) [reply packages/core/src/browser/style/tree.css]: I believe they use the following as a color:

```css
.monaco-tl-indent > .indent-guide {
  transition: border-color 0.1s linear;
}
```

https://github.com/microsoft/vscode/blob/0ae3b979d7847b1f8fdf57ba0aaf1d25484be827/src/vs/base/browser/ui/tree/media/tree.css#L25-L34

File: packages/navigator/src/browser/navigator-widget.tsx (hunk 4 above)
Comment (kaiyue0329) [reply packages/navigator/src/browser/navigator-widget.tsx]: I could not think of a better way to do it at the moment. I was wondering if you have any recommendation.

File: packages/core/src/browser/style/tree.css (hunk 2 above)
Comment (kaiyue0329) [reply packages/core/src/browser/style/tree.css]: I used `var(--theia-foreground)` for the color and it seems to work.

File: packages/navigator/src/browser/navigator-widget.tsx (hunk 4 above)
Comment (akosyakov) [reply packages/navigator/src/browser/navigator-widget.tsx]: I was thinking that we will change rendering and insert as many div block as deep a node. When we call add a border add different depth levels which is nicely aligned vertically for all nodes. So only nodes rendering should be changed. Which level depth level should be decorated should be decided based on some preferences, when such preferences are changed the widget should be updated.

Comment (marechal-p) [reply packages/navigator/src/browser/navigator-widget.tsx]: The minimal modification would be to add as many divs as there is depth **per node row**. The better but more involving way is to change the rendering of the tree to wrap rows of a certain depth together in the same div: it requires rewriting a good chunk of the rendering logic from what I can see. Do you think @kaiyue0329 should look into doing the former or the latter?

Do you have hints for him? I don't want to lead him into a direction you wouldn't approve of, as I am not really familiar with the trees myself.

Comment (marechal-p) [reply packages/navigator/src/browser/navigator-widget.tsx]: Ok, we looked through VS Code's implementation, and they add a bunch of little indent strokes on every node. We'll try to follow this design.

Comment (akosyakov) [reply packages/navigator/src/browser/navigator-widget.tsx]: Minimal is better, trees are used a lot, we cannot break them.

**Comment by kaiyue0329** [comment]
I opened a new PR that adds indent guidelines to all trees in Theia.
